### DISCUSSION TITLE
Question about ADInertialForce

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#24845] [Q&A General]
Hi All,

I am using ADDynamicStressDivergenceTensors to set up a dynamics problem. Something I noticed is that when I use IntertialForce in tandem with ADDynamicStressDivergenceTensors, the problem works fine. However, if I use ADInertialForce, I find that SVD analysis returns singular values(and the NL iterations do not converge). Is this a known bug or am I supposed to use a non-AD inertia kernel?

**Comment by maxnezdyur** [comment]
I have had some problems with this also but I am surprised about the singular values with the SVD. Can't give any answers to your questions, but I have some questions of my own! What does the convergence history look like and what strain are you using?

**Reply by TheGreatCid** [reply]
This is with finite strain and the linear iterations breakdown at the first timestep

**Reply by maxnezdyur** [reply]
Not sure if this will help with yours, but if you have time try the below because it helped me before. I had this weird thing where if it goes past the first NL iteration it worked, but this was a while ago.
```
[Dampers]
  [jac]
    type = ElementJacobianDamper
    max_increment = 0.01
  []
[]
```
Edit: Misread your earlier message as nonlinear iteration broke down, not sure if the Damper would even help with the linear iterations. Also still would want the whole singular value issue looked into even if this "fixed" the problem.

**Reply by TheGreatCid** [reply]
yeah, the damper is nice, but it won't help with singular values. It's very odd...

**Reply by lindsayad** [reply]
If you share a MOOSE-only minimal example I can do some investigation

**Reply by TheGreatCid** [reply]
Bit of an issue there. I did try to create MOOSE-only MWE, but I have been unable to replicate the bug. So I'm probably on my own here.
